app/models/concerns/users/user_manipulator.rb

```diff
@@ -65,6 +69,14 @@ def update_user(params)
       UserEmailChangedMailer.changed_email(user).deliver_now if user.email_updated?
     end
 
+    def destroy_user(params)
+      raise PermissionError unless can_delete_user?
```

Comment:
true, but I tend to like these kinds of checks everywhere... and if it is in 1 place, I would prefer where the action is actually happening in the model